src/ray/gcs/gcs_server/gcs_actor_scheduler.cc

```diff
@@ -563,5 +623,45 @@ void RayletBasedActorScheduler::HandleWorkerLeaseReply(
   }
 }
 
+void GcsActorScheduler::HandleWorkerLeaseRejectedReply(
+    std::shared_ptr<GcsActor> actor, const rpc::RequestWorkerLeaseReply &reply) {
+  // The request was rejected because of insufficient resources.
+  if (!actor->GetAcquiredResources().IsEmpty()) {
+    // Return the actor's acquired resources, which updates GCS' resource view.
+    ReturnActorAcquiredResources(actor);
```

Comment:
If raylet scheduling is used, actor's acquired resources would always be empty. We don't have to worry about it here. Besides, I'm trying to reduce the using of the gcs scheduling feature flag because both schedulers have to coexist in the end.

Reply:
Adding a CHECK or a comment will help future readers: they might not know that `If raylet scheduling is used, actor's acquired resources would always be empty.`